### PR TITLE
Convert some fixed-size arrays to vectors or maps

### PR DESCRIPTION
**Pull request by Noordfrees**
Re #212 

This also fixes a bug with saveloading rides-related data after removing ride instances: Saveloading code assumes that every ride instance has a unique index that never changes, which was not guaranteed until now.

### CODE REVIEW

File: src/coaster_gui.cpp [hunk 1]
```diff
@@ -524,17 +524,17 @@ void CoasterInstanceWindow::OnClick(WidgetNumber widget, const Point16 &pos)
 
 		case CIW_CHOOSE_ENTRANCE: {
 			DropdownList itemlist;
-			for (int i = 0; _rides_manager.entrances[i] != nullptr; i++) {
-				_str_params.SetUint8(1, _language.GetText(_rides_manager.entrances[i]->name));
+			for (auto &eetype : _rides_manager.entrances) {
```
Comment (Alberth289346) [inline src/coaster_gui.cpp]: We got the option to use `const` data here, is that something to aim for? (I'd say yes, but maybe you have other plans?)

File: src/ride_type.cpp [hunk 2]
```diff
@@ -681,18 +681,10 @@ void RideInstance::Save(Saver &svr)
...
-	std::fill_n(this->exits, lengthof(this->exits), nullptr);
 }
 
 RidesManager::~RidesManager()
 {
-	for (uint i = 0; i < lengthof(this->entrances); i++) delete this->entrances[i];
-	for (uint i = 0; i < lengthof(this->exits); i++) delete this->exits[i];
-	for (uint i = 0; i < lengthof(this->ride_types); i++) delete this->ride_types[i];
-	for (uint i = 0; i < lengthof(this->instances); i++) delete this->instances[i];
 }
```
Comment (Alberth289346) [inline src/ride_type.cpp]: Constructor can be dropped. Destructor can be dropped too, it doesn't look like a baseclass at least.

File: src/coaster_gui.cpp (hunk 1 above)
Comment (Alberth289346) [reply src/coaster_gui.cpp]: You seem to use `const` at other places, so this one is likely a bad suggestion.

Comment (Noordfrees) [reply src/coaster_gui.cpp]: As a rule I try to use `const` wherever possible – it's a good way to discover bugs already at compile-time. Only variables that are actually meant to be modified should be non-`const`.

**Pull request merged**